packages/ui/react-ui-deck/src/components/Deck/Deck.tsx

```diff
@@ -50,20 +49,24 @@ const deckGrid =
 
 // TODO(thure): `justify-center` will hide some content if overflowing, nor will something like `dialogLayoutFragment` containing the Deck behave the same way.
 //  Currently `justify-center-if-no-scroll` is used, which relies on support for `animation-timeline: scroll(inline self)`, which is not broad.
-const deckLayout = `overflow-y-hidden snap-inline snap-proximity sm:snap-none sm:justify-center-if-no-scroll ${deckGrid}`;
+const deckLayout = `overflow-y-hidden overflow-x-auto snap-inline snap-proximity sm:snap-none sm:justify-center-if-no-scroll ${deckGrid}`;
+
+const soloLayout =
+  'grid grid-rows-[var(--rail-size)_[toolbar-start]_var(--rail-action)_[content-start]_1fr_[content-end]] grid-cols-1';
 
 const resizeButtonStyles = (...etc: ClassNameValue[]) =>
   mx(resizeHandle, resizeHandleVertical, 'hidden sm:grid row-span-3', ...etc);
 
 const DeckRoot = forwardRef<HTMLDivElement, DeckRootProps>(
-  ({ classNames, children, asChild, overscroll, solo, ...props }, forwardedRef) => {
+  ({ classNames, children, asChild, solo, ...props }, forwardedRef) => {
     const Root = asChild ? Slot : 'div';
+
     return (
-      <DeckProvider overscroll={overscroll} solo={solo}>
+      <DeckProvider solo={solo}>
         <Root
           {...props}
-          style={{ ...props.style, containerType: 'inline-size' }} // TODO(Zan): We should have a class for this?
-          className={mx(deckLayout, classNames, solo ? 'overflow-x-hidden' : 'overflow-x-auto')}
+          style={{ ...props.style }} // TODO(Zan): We should have a class for this?
```

Comment:
Leave your comment but remove setting the `style` prop, it will get this from the props spread in the line above.